dbt/config.py

```diff
@@ -355,6 +363,43 @@ def from_current_directory(cls, cli_vars):
     def hashed_name(self):
         return hashlib.md5(self.project_name.encode('utf-8')).hexdigest()
 
+    def get_resource_config_paths(self):
+        """Return a dictionary with 'seeds' and 'models' keys whose values are
+        lists of lists of strings, where each inner list of strings represents
+        a configured path in the resource.
+        """
+        return {
+            'models': _get_config_paths(self.models),
+            'seeds': _get_config_paths(self.seeds),
+        }
+
+    def get_unused_resource_config_paths(self, resource_fqns):
+        """Return a list of lists of strings, where each inner list of strings
+        represents a type + FQN path of a resource configuration that is not
+        used.
+        """
+        resource_config_paths = self.get_resource_config_paths()
+        unused_resource_config_paths = []
+        for resource_type, config_paths in resource_config_paths.items():
+            fqns = resource_fqns.get(resource_type, [])
+            for config_path in config_paths:
+                if not _is_config_used(config_path, fqns):
+                    unused_resource_config_paths.append(
+                        [resource_type] + config_path
+                    )
+        return unused_resource_config_paths
+
+    def warn_for_unused_resource_config_paths(self, resource_fqns):
+        unused = self.get_unused_resource_config_paths(resource_fqns)
```

Comment:
There's an unfortunate effect that happens here:

Disabling models in `dbt_project.yml` file will effectively "remove" the models from the manifest, therefore causing dbt to think that the config is useless. 

For example:
```
models:
    my_project:
            enabled: false
```

If your project name is `my_project`, this will (errantly) result in:
```
WARNING: Configuration paths exist in your dbt_project.yml file which do not apply to any resources.
There are 1 unused configuration paths:
- models.my_project
```

I don't really know that there's a good fix for this, as we've consciously decided to discard disabled resources in parsing. Would it be crazy to skip adding paths in `get_unused_resource_config_paths` if there's an `enabled: false` present? I think that would probably get pretty messy....

Reply:
Ok, I got this done. I had to refactor the loader code quite a bit, the `ResourceLoader` abstraction ended up getting in the way quite a bit, so now the `GraphLoader` just explicitly builds up a manifest as it goes. I think if we want to do more things like this (retaining extra data as we load/parse) we'll have to refactor the parser/loader code further.